migrations/2024-04-17-084906_add_generic_link_table/up.sql

```diff
@@ -0,0 +1,18 @@
+CREATE TYPE "GenericLinkType" as ENUM(
+    'payment_method_collect',
+    'payout_link'
+);
+
+CREATE TABLE generic_link (
+  link_id VARCHAR (64) NOT NULL PRIMARY KEY,
+  primary_reference VARCHAR (64) NOT NULL,
+  merchant_id VARCHAR (64) NOT NULL,
+  created_at timestamp NOT NULL DEFAULT NOW():: timestamp,
+  last_modified_at timestamp NOT NULL DEFAULT NOW():: timestamp,
+  expiry timestamp NOT NULL DEFAULT NOW():: timestamp,
+  link_data JSONB NOT NULL,
+  link_status VARCHAR (256) NOT NULL,
+  link_type "GenericLinkType" NOT NULL,
+  url VARCHAR (256) NOT NULL,
+  return_url VARCHAR(256) NULL
```

Comment:
@Narayanbhat166 Considering that `return_url` in the `business_profile` table is `TEXT`, should we keep these fields as `VARCHAR(255)` or `TEXT`?